cli/cmd/vuln_host.go

```diff
@@ -1046,52 +1170,42 @@ func hostScanPackagesVulnPackagesTable(pkgs filteredPackageTable) [][]string {
 	return out
 }
 
-func removeFixedVulnerabilitiesFromAssessment(assessment *api.HostVulnHostAssessment) {
-	out := []api.HostVulnCVE{}
-
-	for _, cve := range assessment.CVEs {
-		var filteredCves []api.HostVulnPackage
-		for _, pkg := range cve.Packages {
-			if pkg.VulnerabilityStatus == "Fixed" {
-				continue
-			}
-			filteredCves = append(filteredCves, pkg)
-		}
-		cve.Packages = filteredCves
-		if len(cve.Packages) > 0 {
-			out = append(out, cve)
+func removeFixedVulnerabilitiesFromAssessment(assessment []api.VulnerabilityHost) []api.VulnerabilityHost {
+	var filteredCves []api.VulnerabilityHost
+	for _, cve := range assessment {
+		if cve.Status != "Fixed" {
+			filteredCves = append(filteredCves, cve)
 		}
 	}
-	assessment.CVEs = out
+	return filteredCves
 }
 
 // Build the cli output for vuln host show-assessment
-func buildVulnHostReports(assessment api.HostVulnHostAssessment) error {
+func buildVulnHostReports(response api.VulnerabilitiesHostResponse) error {
 	// @afiune the UI today doesn't display any vulnerability that has been fixed
 	// but the APIs return them, this is causing confusion, to fix this issue we
 	// are filtering all of those "Fixed" vulnerabilities here
-	removeFixedVulnerabilitiesFromAssessment(&assessment)
+	response.Data = removeFixedVulnerabilitiesFromAssessment(response.Data)
 
-	hostVulnCounts := assessment.VulnerabilityCounts()
+	hostVulnCounts := response.VulnerabilityCounts()
 	if hostVulnCounts.Total == 0 {
 		if cli.JSONOutput() {
-			return cli.OutputJSON(assessment)
+			return cli.OutputJSON(response.Data)
 		}
 		cli.OutputHuman("Great news! This host has no vulnerabilities... (time for %s)\n", randomEmoji())
 		return nil
 	}
 
 	var (
-		mainReport                  = hostVulnHostDetailsMainReportTable(assessment)
-		filteredCves, filtered      = filterHostCVEsTable(assessment.CVEs)
+		mainReport                  = hostVulnHostDetailsMainReportTable(response)
+		filteredCves, filtered      = filterHostCVEsTable(cvesSummary(response.Data))
 		detailsReport               = buildVulnHostsDetailsTable(filteredCves)
 		csvHeader, csvDetailsReport = buildVulnHostsDetailsTableCSV(filteredCves)
 	)
-	assessment.CVEs = filteredCves
```

Comment:
this is a regression since we are not filtering the JSON output and before we were doing it here.